### PR TITLE
Add typescript, fix bugs

### PR DESCRIPTION
**Pull request by jxjj**
- Convert a number of components to typescript and fix the resulting bugs.

- Refactor Trekker's Map components to make things a bit more resilient. There were a slew a cases that would cause the app to white page if the tour had missing routes, target points, and starting locations. Now, when getting a stop target, `getStopPointAtIndex` will now work backwards if a targetPoint is not defined: trying previous stops, the start_location, and ultimately using a default point (West Bank Plaza at UMN) if none exist.

- Refactor Pinia store to use a function instead of `this`. Pinia does a bit of unwrapping magic when using state and getters. If you want to pass an argument to a getter, you have to be careful how to write the function so that it doesn't bind `this` incorrectly.  It was easy to overlook and hard to catch errors. Instead, I've opted for the less magical approach: getters just use computed under the hood, so we write our getters using computed. It's a bit more boilerplate, but easier to understand IMO.

- Remove composables (aka 'hooks'), and use the Pinia store directly to get state or dispatch actions.

- More active sheet setting into the Trekker store so that when a user clicks on a Map Popup button to go to a stop, the open MapSheet will close and navigate to the correct page. (Previously, it would route to the correct page, but the user wouldn't be able to tell because the map sheet remained open.

- Fix some tests and add a few more for the TourStopPage in Camino Creator.

- Update tour seeders, removing references to the old hotlinks feature.

- Change the max upload size in sail's nginx web container to 50MB. (Was previously the default 1MB).

### CODE REVIEW

File: resources/camino-trekker/App.vue [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <script setup>
-import { useCreatorStore } from "@/camino-creator/stores/useCreatorStore";
+// import { useCreatorStore } from "@/camino-creator/stores/useCreatorStore";
```
Comment (cmcfadden) [inline resources/camino-trekker/App.vue:2]: nuke this whole block?

File: resources/camino-trekker/components/Stage/stages/Guide.vue [hunk 2]
```diff
@@ -4,22 +4,18 @@
...
-    default: "en",
-  },
-});
-const markdown = computed(() => props.stage.text[props.locale]);
+interface Props {
+  stage: GuideStage;
+}
+const props = defineProps<Props>();
+const store = useTrekkerStore();
+const markdown = computed(() => props.stage.text?.[store.locale] ?? "");
```
Comment (cmcfadden) [inline resources/camino-trekker/components/Stage/stages/Guide.vue:18]: I still think this syntax is witchcraft.

File: resources/camino-trekker/stores/useTrekkerStore.ts [hunk 3]
```diff
@@ -1,87 +1,106 @@
...
-      if (this.isLastStop) return null;
-      return this.allStops[this.stopIndex + 1];
-    },
-    previousStop(): Maybe<TourStop> {
-      if (this.isFirstStop) return null;
-      return this.allStops[this.stopIndex - 1];
-    },
-  },
-  actions: {
+  // STATE
```
Comment (cmcfadden) [inline resources/camino-trekker/stores/useTrekkerStore.ts:19]: MINNESOTA

File: resources/camino-trekker/App.vue (hunk 1 above)
Comment (jxjj) [reply resources/camino-trekker/App.vue:2]: Good call – thanks!

**Pull request merged**